src/wmtk/invariants/MinIncidentValenceInvariant.cpp

```diff
@@ -0,0 +1,37 @@
+#include "MinIncidentValenceInvariant.hpp"
+
+#include <wmtk/Mesh.hpp>
+#include <wmtk/SimplicialComplex.hpp>
+
+namespace wmtk::invariants {
+
+
+MinIncidentValenceInvariant::MinIncidentValenceInvariant(const Mesh& m, long min_valence)
+    : MeshInvariant(m)
+    , m_min_valence(min_valence)
+{}
+
+bool MinIncidentValenceInvariant::before(const Tuple& t) const
+{
+    const Tuple v0 = t;
+    const Tuple v1 = mesh().switch_vertex(t);
+    const long val0 = static_cast<long>(SimplicialComplex::vertex_one_ring(mesh(), v0).size());
```

Comment:
lets not use the SimplicialComplex interface actively in new code

Reply:
#478